lib/monitor.js

```diff
@@ -45,6 +46,41 @@ function defaultIsErrorCallback(status) {
     return (/^([45].+$).*/.exec(status)) != null
 }
 
+/**
+ * Get error message from the response. If error message is null, sets the string to empty. 
+ * @param {HTTP response} res the http response
+ * @returns a string with the error message or empty string if error message not found.
+ */
+function getErrorMessage(res){
+    return res.get("Error-Message") ? res.get("Error-Message") : ""
+}
+
+/**
+ * Collect latency metric dependency_request_seconds 
+ * @param {HTTP resquest} req the http request
+ * @param {HTTP response} res the http response
+ * @param {string} name the name of dependency
+ * @param {string} type which request protocol was used (e.g. http, grpc, etc)
+ */
+function collectDependencyTime(req, res, name, type) {
+    end = dependencyRequestSeconds.startTimer()
+    var isErr = defaultIsErrorCallback(res.statusCode);
+    var errorMsg = getErrorMessage(res)
+
+    // observes the dependency request duration
+    res.once("finish", () => {
+        end({
```

Comment:
@KarineValenca, it's possible that other parts of the code also adds some promise to the `"finish"` hook... did you test if your code copes well in such scenario?

For example, assuming the `res` object is the same, would this call to `end()` work well with the call to `end()` on `Monitor.init()`'s line 140`?

Reply:
It will work perfectly depending on the Node version but needs further investigation to guarantee that it works on other versions.

Reply:
Ok. Will assume the risk right now. Thanks!